### PR TITLE
Softmax with Cross Entropy operator migration and pass communication initialization fix

### PR DESCRIPTION
**Pull request by Function-Samuel**
### PR Category
Communication Library

### PR Types
Others

### Description
softmax with cross entropy算子迁移与pass的通信初始化修复
PCard-85979

### CODE REVIEW

File: python/paddle/distributed/passes/auto_parallel_replace_with_parallel_cross_entropy.py [hunk 1]
```diff
@@ -57,9 +60,13 @@ def _check_user(self, value):
     def _apply_single_impl(self, main_program, startup_program, context):
         del_ops = []
         new_ops = []
-        ring_id = self.model_parallel_group.id
-        rank = self.model_parallel_group.rank
-        nranks = self.model_parallel_group.nranks
+
+        params = main_program.global_block().all_parameters()
+        process_ids = params[0].dist_attr().process_mesh.process_ids
```
Comment (liym27) [inline python/paddle/distributed/passes/auto_parallel_replace_with_parallel_cross_entropy.py]: Why use `process_ids` of params[0] instead of the input of 'pd_op.cross_entropy_with_softmax'?

Comment (Function-Samuel) [reply python/paddle/distributed/passes/auto_parallel_replace_with_parallel_cross_entropy.py]: get it, I will modify this immediately

File: python/paddle/distributed/passes/auto_parallel_replace_with_parallel_cross_entropy.py [hunk 2]
```diff
@@ -73,6 +73,12 @@ def _apply_single_impl(self, main_program, startup_program, context):
                     if not placement1[1].is_shard():
                         return
 
+                    process_ids = op.dist_attr.process_mesh.process_ids
```
Comment (liym27) [inline python/paddle/distributed/passes/auto_parallel_replace_with_parallel_cross_entropy.py]: Why use process_ids from `op.dist_attr.process_mesh` instead of  process_ids from 'operand1'?

Comment (Function-Samuel) [reply python/paddle/distributed/passes/auto_parallel_replace_with_parallel_cross_entropy.py]: My understanding of this is in paddle computation graph's value is an edge and op is a node, I have considered that the process_ids in op and operand might be different as well. So I guess the process_ids obtained from op may be suitable here.

**Review by liym27 — APPROVED** [review]
LGTM

**Review by zhangbo9674 — APPROVED** [review]
LGTM
